src/Platforms/SQLServer/Comparator.php

```diff
@@ -23,6 +24,17 @@ public function __construct(SQLServerPlatform $platform, string $databaseCollati
         $this->databaseCollation = $databaseCollation;
     }
 
+    public function compareTables(Table $fromTable, Table $toTable): TableDiff
+    {
+        $fromTable = clone $fromTable;
+        $toTable   = clone $toTable;
+
+        $this->normalizeColumns($fromTable);
```

Comment:
We should probably make `normalizeColumns` perform the cloning and return a new instance (as done in the MySQL comparator) to make code less subject to bugs where we would mutate the external object by mystake